### ISSUE TITLE
amf.apicontract.client.platform.model.domain.Tag.withVariables seems misnamed

### ISSUE THREAD

**Issue by jgsuess** [aml-org/amf#1425]
Your issue may already be reported! Please search on Github issues before creating one.

* **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] question
  - [ ] other

* **What is the current behavior?**

The method for setting the documentation for a Tag is called `withVariables` and documented as *setting the host property for the server*. This does not seem to correspond with the action in the code. See also semantic definition in [Tag](https://github.com/aml-org/amf/blob/develop/documentation/model.md#tag).

```scala
  /** Set host property of this Server. */
  def withVariables(documentation: CreativeWork): this.type = {
    _internal.withDocumentation(documentation)
    this
  }
```

* **If the current behavior is a bug, please provide the minimal steps to reproduce with AMF code. Please don't upload AMF code mixed with yours.**

* **What is the expected behavior?**
Method to be called `withDocumentation`

* **What is the motivation/use case for changing the behavior? (for feature requests)**
Automated reflective transcription

* **Please tell us about your environment:**
  
  - AMF Version: 5
  - AMF Distribution: JVM
  - Operating System: Ubuntu 20.04
  - Link to your project: proprietary

* **Other information** (e.g. detailed explanation, stack traces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, Gitter, etc)

**Comment by arielmirra** [comment]
Hi @jgsuess! That's correct, the naming is not on point. To be backward compatible, we'll not delete that method but add a new `withDocumentation()` one.

Thank you for reporting, I've created an internal issue to track this, with id W-11233146. I'll keep you up to date, it should be available in the AMF release version 5.0.10, next month.

**Comment by arielmirra** [comment]
Hi @jgsuess! This issue has been solved in PR #1454! It'll be available in the next AMF release (coming after 5.0.10). Thank you!
